### DISCUSSION TITLE
cloud liquid water path in SCM output

### DISCUSSION THREAD

**Discussion by evankalina** [NCAR/ccpp-scm#281] [General]
Hi @grantfirl and others,

I have some SCM output and am looking at the variables rad_cloud_iwp, rad_cloud_lwp, rad_cloud_rwp, and rad_cloud_swp. The netcdf output file describes them as water _paths_, with units of g/m2. But they are 2D fields, with dimensions of time x height. Are they actually water _contents_ with units of g/m3? My understanding is that the water path is the vertical integral of the water content. Thanks for any clarification that you can provide.

If you would like to see my output, it is here (on Orion):
/work/noaa/ufs-phys/ekalina/ccpp-scm-20210816/scm/run/output_LASSO_20180514_s02_SCM_GFS_v16/output.nc

Cheers,
evan

**Comment by grantfirl** [comment]
Hi @evankalina ,

The variables that are written out in the SCM output come from a multi-field array holding cloud property diagnostics used by radiation. It is simply named "clouds" in the code and is in the interstitial GFS DDT. This array is populated via one of several subroutines in the module_radiation_clouds module (see https://github.com/NCAR/ccpp-physics/blob/main/physics/radiation_clouds.f) depending on which microphysics and radiation combination is being used within a suite. These subroutines translate the meteorological state from the rest of the model into variables that the radiation expects in preparation for solving the actual radiative transfer problem. All of this to say that these variables are indeed water paths with units g m-2, but they are valid for a given level, which is what the radiation apparently expects; i.e. integrated between the vertical interfaces of a level. I think that in order to get the integrated diagnostic that you would expect for the entire column, you could simply sum over the vertical index.

**Comment by dustinswales** [comment]
@evankalina 
Everything @grantfirl said above applies to the RRTMGP radiation, except there is no "clouds" array, but rather Interstitials for each of the cloud radiative properties (e.g for the GP to GFDL coupling https://github.com/NCAR/ccpp-physics/blob/main/physics/GFS_rrtmgp_gfdlmp_pre.F90#L80)

**Comment by evankalina** [comment]
@grantfirl and @dustinswales Thanks so much for your quick responses. Based on your explanation, I agree that we should be able to sum over the height dimension to get the total water path. Thank you!
